### PR TITLE
Move job_details region under info_panel

### PR DESCRIPTION
**Pull request by rbillings**
Tests pass except test_unclassified_job.py. This gives an error on switching the window that I don't know how to fix- the logviewer page does open correctly.

```
    def open_logviewer(self):
        self.find_element(*self._job_details_panel_locator).send_keys('l')
        window_handles = self.selenium.window_handles
        for handle in window_handles:
            self.selenium.switch_to.window(handle)
>       return LogviewerPage(self.selenium, self.page.base_url).wait_for_page_to_load()
E       AttributeError: 'InfoPanel' object has no attribute 'base_url'

pages/treeherder.py:414: AttributeError
------------------------------------------- pytest-selenium --------------------------------------------
URL: https://treeherder.allizom.org/logviewer.html#?job_id=78079209&repo=mozilla-inbound
================================= 1 failed, 2 passed in 17.98 seconds ==================================
ERROR: InvocationError: '/Users/rebeccabillings/github/treeherder-tests/.tox/tests/bin/pytest --junit-xml=results/tests.xml --html=results/tests.html --base-url https://treeherder.allizom.org tests/test_unclassified_job.py'
```

### CODE REVIEW

File: pages/treeherder.py [hunk 1]
```diff
@@ -378,38 +373,50 @@ def title(self):
...
-            for handle in window_handles:
-                self.selenium.switch_to.window(handle)
-            return LogviewerPage(self.selenium, self.page.base_url).wait_for_page_to_load()
-
-        def pin_job(self):
-            el = self.find_element(*self._job_keyword_locator)
-            self.wait.until(EC.visibility_of(el))
-            self.find_element(*self._pin_job_locator).click()
+        def job_details(self):
+            return self.JobDetails(self)
```
Comment (davehunt) [inline pages/treeherder.py]: The `Region` constructor expects a `Page` and `self` is a `Region`. Change this to `return self.JobDetails(self.page)` and your failure should go away.

File: pages/treeherder.py [hunk 2]
```diff
@@ -378,38 +373,50 @@ def title(self):
...
+
+        class JobDetails(Region):
+
+            _job_details_panel_locator = (By.ID, 'job-details-panel')
```
Comment (davehunt) [inline pages/treeherder.py]: We should have a root locator if possible, and looking at the DOM, I think we could just rename this locator to `_root_locator`.

File: pages/treeherder.py [hunk 3]
```diff
@@ -378,38 +373,50 @@ def title(self):
...
+
+            @property
+            def job_result_status(self):
+                return self.find_element(*self._job_result_status_locator).text
+
+            def filter_by_job_keyword(self):
+                self.find_element(*self._job_keyword_locator).click()
+
+            def open_logviewer(self):
+                self.find_element(*self._job_details_panel_locator).send_keys('l')
```
Comment (davehunt) [inline pages/treeherder.py]: With the root locator change, this would be `self.root.send_keys('l')`

**Comment by rbillings** [comment]
Rebased & updated with recommended changes. Ready for review @mozilla/firefox-test-engineering.

**Pull request merged**